nipype/interfaces/base.py

```diff
@@ -1452,12 +1448,30 @@ def _process(drain=0):
             result['merged'] = result['stdout']
             result['stdout'] = []
     else:
-        proc.communicate()  # Discard stdout and stderr
+        stdoutstr, stderrstr = proc.communicate()
+        if output == 'allatonce':  # Discard stdout and stderr otherwise
+            result['stdout'] = read_stream(stdoutstr, logger=iflogger)
+            result['stderr'] = read_stream(stderrstr, logger=iflogger)
+        del stdoutstr
+        del stderrstr
```

Comment:
Sorry, meant for these to replace the `del stdout` and `del stderr` below, since `stdoutstr` and `stderrstr` can be created in the other branch.

I guess, though, that those only exist under certain sub-branches. Not sure how you want to go about this.